htdocs/core/js/lib_head.js.php

```diff
@@ -1340,8 +1340,8 @@ function price2numjs(amount) {
 	// Close drop down
 	jQuery(document).on("click", function(event) {
 		// search if click was outside drop down
-		if (!$(event.target).closest('.butAction.dropdown-toggle').length) {
-			let parentholder = jQuery(".butAction.dropdown-toggle").closest(".dropdown.open");
+		if (!$(event.target).closest('.dropdown-toggle').length) {
```

Comment:
In PR #30835, a contributor has fixed by doing
`var parentholder = jQuery(event.target).parent();`
instead of removing the .butAction
`var parentholder = jQuery(".butAction.dropdown-toggle").closest(".dropdown");`

Which one is better between this change the one of the #30835

Reply:
`var parentholder = jQuery(event.target).parent();`

was my first fix before I change to 
`if (!$(event.target).closest('.xxxx.dropdown-toggle').length) {`
and then to 
`if (!$(event.target).closest('.dropdown-toggle').length) {`

the difference is subtle. but in the first case, the closure is strongly dependent on the semantic structure of the dom and will therefore be less robust to future changes.

exemple of evolution bloqued :

Current dom of buttons :
```html
<div class="dropdown inline-block dropdown-holder">
  <a style="margin-right: auto;" class="dropdown-toggle classfortooltip butAction" title="" data-toggle="dropdown">Créer</a>
  <div class="dropdown-content">
    <a class="butAction" href="/thersane/gestion/htdocs/fourn/commande/card.php?action=create&amp;origin=commande&amp;originid=24888" title="" aria-label="">Créer commande d'achat</a>
    <a class="butAction" href="/thersane/gestion/htdocs/expedition/shipment.php?id=24888" title="" aria-label="">Créer expédition</a>
    <a class="butAction" href="/thersane/gestion/htdocs/compta/facture/card.php?action=create&amp;token=b05512f53c46467d0c30107eb95219c9&amp;origin=commande&amp;originid=24888&amp;socid=8144" title="" aria-label="">Créer Facture</a>
  </div>
</div>```

exemple of implementation of sub menu :
```html
<div class="dropdown inline-block dropdown-holder">
  <a style="margin-right: auto;" class="dropdown-toggle classfortooltip butAction" title="" data-toggle="dropdown">Créer</a>
  <ul class="dropdown-content">
    <li>
      <a class="butAction" href="#">Créer commande d'achat</a>
    </li>
    <li>
      <a class="butAction" href="#">Créer expédition</a>
    </li>
    <li>
      <a class="butAction" href="#">Créer Facture</a>
    </li>
    <li class="dropdown-separator">Title of separator</li>
    <li class="dropdown-sub">
          <a class="button-to-open-sub-menu-exanple" href="#">Sub menu </a>
               <ul class="dropdown-content">
                  <li>
                        <a class="butAction" href="#">sub menu action 01</a>
                  </li>
                  <li>
                        <a class="butAction" href="#">sub menu action 02</a>
                  </li>
                  <li>
                        <a class="butAction" href="#">sub menu action 02</a>
                  </li>
      </ul>
    </li>
</div>```

So with the code 
`var parentholder = jQuery(event.target).parent();`

By click on link to open submenu the menu will close.

But in the other case 
`var parentholder = jQuery(".butAction.dropdown-toggle").closest(".dropdown");`
javascript will walkthrough the DOM hierarchy to find out whether we're in a drop-down list or not.

Reply:
Hi!

He  fixed the open behavior **not** the close behavior.
We dont fix the same thing.

![image](https://github.com/user-attachments/assets/0ce420e4-d7bb-48cc-b16b-9fa4891c6150)

Reply:
in PR #30835, I probably would have done something like this:

```javascript
var parentholder = jQuery(event.target).closest(".dropdown");
```
instead of

```javascript
var parentholder = jQuery(event.target).parent();
```
because use closest is less dependent on DOM I think but use more CPU...

But I haven't tested it, so I can't say which is best.